### PR TITLE
Move examples from the cloud-storage-docs-xml-api-examples repo, and update them to use Application Default Credentials.

### PR DESCRIPTION
**Pull request by jerjou**

### CODE REVIEW

**Comment by waprin** [comment]
LGTM, but one small nit, for the cmd-line sample README can you add a link back to the doc page that includes from it?

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+//[START all]
```
Comment (tmatsuo) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: If embedded in the doc, will the license header be in the snippet?

File: cloud-storage/xml-api/cmdline-sample/key.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Please download key.json file from the Google APIs Console at https://console.developers.google.com.
```
Comment (tmatsuo) [inline cloud-storage/xml-api/cmdline-sample/key.json]: Why do you want to have this file? Can you just add it to the README?

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  private static final HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+
+  /** Global instance of HTML reference to XSL style sheet. */
+  String XSL = "\n<?xml-stylesheet href=\"/xsl/listing.xsl\" type=\"text/xsl\"?>\n";
+
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+
+    try {
+      AppIdentityCredential credential = new AppIdentityCredential(Arrays.asList(STORAGE_SCOPE));
```
Comment (tmatsuo) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: I think you can also use GoogleCredential.getApplicationDefault() here. Is there a reason not using it?

File: cloud-storage/xml-api/cmdline-sample/src/main/java/StorageSample.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
+//[START all]
```
Comment (tmatsuo) [inline cloud-storage/xml-api/cmdline-sample/src/main/java/StorageSample.java]: I think we should have a coding standard and the current consensus is to use the sun's one, according to:
https://github.com/GoogleCloudPlatform/Template/wiki/style.html

What do you think?

**Comment by waprin** [comment]
@tmatsuo good point I merged master so we could get the root pom in here and added the checkstyle plugin which docs are telling me defaults to sun. Need to fix some issues before it will pass on travis now though.

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java (hunk 1 above)
Comment (jerjou) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: I guess so. Is the 'all' tag used for anything? Perhaps it'd be prudent to just remove the tag?

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java (hunk 3 above)
Comment (jerjou) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: Yeah - I tried doing it, and it wasn't finding the method for some reason. I looked into it a bit, but didn't make much headway, so I put it off so I could at least get the cmdline sample updated.

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java (hunk 1 above)
Comment (tmatsuo) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: Yeah, if nobody uses it, maybe you can just remove it.

File: cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java (hunk 3 above)
Comment (tmatsuo) [inline cloud-storage/xml-api/serviceaccount-appengine-sample/src/main/java/com/google/api/client/sample/storage/appengine/serviceaccount/StorageSample.java]: You need a newer version of com.google.api-client. Maybe you can try [1.20.0](http://search.maven.org/#artifactdetails%7Ccom.google.api-client%7Cgoogle-api-client%7C1.20.0%7Cjar).

**Comment by jerjou** [comment]
Fixed style issues in cloud-storage.

File: taskqueue/deferred/src/main/java/com/google/cloud/taskqueue/samples/DeferSampleServlet.java [hunk 5]
```diff
@@ -42,15 +53,20 @@ public void run() {
...
+    /**
+     * Basic demonstration of adding a deferred task.
+     * @param request servlet request
+     * @param resp servlet response
+     */
     @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse
-            resp) throws IOException {
+    public void doGet(final HttpServletRequest request, final
+    HttpServletResponse resp) throws IOException {
```
Comment (tmatsuo) [inline taskqueue/deferred/src/main/java/com/google/cloud/taskqueue/samples/DeferSampleServlet.java]: How about to fold just before the second `final`?

**Comment by tmatsuo** [comment]
Now I have just a nit
LGTM

To tell the truth, usable auth with Java dev app server is currently broken, so let's do that with another CL, after it is fixed.

**Comment by tmatsuo** [comment]
You may want to squash before merging the PR. I prefer squashing, but it's up to you @waprin

**Comment by waprin** [comment]
Hmm I don't care but I lean against a merge since a lot of those were self-contained commits and if we squash it will be one huge gigantic commit. I am not against squashing but if it's up to me I prefer not to.

**Comment by tmatsuo** [comment]
@waprin Alright, don't squash then. Thanks for volunteering to fix the style!

**Comment by jerjou** [comment]
Added a commit to make BigqueryUtils not final, because AsyncQuerySample was extending it. Okay to merge if travis passes?

**Comment by waprin** [comment]
@tmatsuo  suggested using getApplicationDefaults in appengine/serviceaccount/StorageSample.java I still think that needs to be done? Or did I miss convo that it does'nt.

**Comment by jerjou** [comment]
Six comments up - apparently usable auth doesn't work with the java dev server yet, so we'll defer it to another PR.

**Comment by tmatsuo** [comment]
Yup I am happy

**Pull request merged**